### PR TITLE
Feat/#442 implement error insufficient balance in callop

### LINKED ISSUE #442: Implement `ErrorInsufficientBalance` in callop
Check circuit implementation for more details, https://github.com/privacy-scaling-explorations/zkevm-circuits/pull/998

### PR DESCRIPTION
**Pull request by KimiWu123**
fixed #442

### CODE REVIEW

File: src/zkevm_specs/evm_circuit/execution/callop.py
Comment (rrtoledo) [inline src/zkevm_specs/evm_circuit/execution/callop.py:35]: I'm not sure to understand what you want to do here. Is that an oversight?

**Review by rrtoledo — COMMENTED** [review]
Except for one word_to_address LGTM!

File: src/zkevm_specs/evm_circuit/execution/callop.py
Comment (KimiWu123) [reply src/zkevm_specs/evm_circuit/execution/callop.py]: Nice catch! Fixed in [dec73b2](https://github.com/privacy-scaling-explorations/zkevm-specs/pull/465/commits/dec73b295494a00caf23fba24e42bbad14880fbc)

**Review by rrtoledo — APPROVED** [review]
LGTM! :)

File: tests/evm/test_callop.py
Comment (ChihChengLiang) [inline tests/evm/test_callop.py:136]: Can you expand on why we are removing these tests?

File: tests/evm/test_callop.py
Comment (ChihChengLiang) [inline tests/evm/test_callop.py:262]: The logic can be simplified.

```suggestion
    is_success = (is_precheck_ok and callee != CALLEE_WITH_REVERT_BYTECODE)
```

**Review by ChihChengLiang — APPROVED** [review]
LGTM after comments are addressed.

File: src/zkevm_specs/evm_circuit/util/call_gadget.py
Comment (KimiWu123) [reply src/zkevm_specs/evm_circuit/util/call_gadget.py:70]: This is a bad naming and I haven't found a better name. `IS_SUCCESS_CALL` is to distinguish `callop` and `error_oog_call`. In our circuit it delcared like this 
```
pub(crate) struct CommonCallGadget<F, const IS_SUCCESS_CALL: bool>
```
That's also the reason why `IS_SUCCESS_CALL` is all capital.

File: tests/evm/test_callop.py
Comment (KimiWu123) [reply tests/evm/test_callop.py:136]: There are 768 test cases in `test_callop` now and that's why I merged these 5 different stacks into 2 to reduce the number of cases. There will be 1,344 cases if I don't merge them. But the most important is I think it still covers the testing scenarios or scope as it was.

**Pull request merged**
